documents/plan.md

```diff
@@ -2,10 +2,23 @@
 
 ## Language evaluation
 
+I will be very happy with my language if I can write a program that allows me to at least specify my preference for a professor, time, and class in a language that looks destinctly different from Prolog. It is going to take a lot of work just to get comfortable with Prolog and figure out what resources their are in Prolog to help with language developement, but I am using Prolog, because I think it will be very interesting to find out. I will try to write extensive tests as i develope a more concrete idea of how each method will be implemented for both the Prolog and the DSL language to ensure they work as I expect them too. I would like to test my language design by asking my critiquers and maybe several people in my dorm to attempt to write a short program with some basic instructions and ask them how they feel about it and see how many times they need my help to figure something out.  
+
 ## Implementation plan
 
+I have already decided to use Prolog as my host language. If I can stay on schedule I would like to have these things done by the given dates:
+
+   + 4/3 Have an example program in Prolog with professor, class, and time preferences working, with me directly inputting class information. (This is likely to be the hardest goal to accomplish, so I may not finish by 4/3 but I should have this and the stuff planned for 4/10 done by 4/10)
+   + 4/10 Design Initial DSL Syntax and write example programs.
+   + 4/17 Have example programs working with syntax modified as necessary.
+   + 4/24 Write something to read in the class information and break it up into all the relevant information.
+
 ## Contingency plan
 
+It will take some time and work to become fluent enough in prolog, so I am going to read some prolog books and attempt to write a program in prolog which does what my DSL should do and then search for ways to remove the host language flavor and make it easier to say what I want to say in my DSL. Technically this is a very ambitious project because I don't really know what tools are available in Prolog for writing DSLs and since I am using Prolog their will be fewer other students who can help me debug if I run into problems, but I will seek out Prof. Ben or Prof. Keller for problems with Prolog specifically and try to talk with and work with other students writing their DSLs for more general questions that aren't domain specific which they may be working through at the same time that I am.
+
```

Comment:
It might be also useful to think about when it's a good time to move on from a particular problem -- how and when to get unstuck, if necessary.